onnxruntime/core/graph/graph_flatbuffers_utils.cc

```diff
@@ -0,0 +1,406 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "graph_flatbuffers_utils.h"
+#include "core/framework/tensorprotoutils.h"
+
+using namespace ONNX_NAMESPACE;
+using namespace ::onnxruntime::common;
+using namespace ::onnxruntime::experimental;
+
+namespace onnxruntime {
+namespace experimental {
+namespace utils {
+
+#if !defined(ORT_MINIMAL_BUILD)
+static flatbuffers::Offset<fbs::Dimension> GetTensorDimensionOrtFormat(
+    flatbuffers::FlatBufferBuilder& builder,
+    const TensorShapeProto_Dimension& tensor_shape_dim) {
+  auto denotation = builder.CreateString(tensor_shape_dim.denotation());
+  flatbuffers::Offset<fbs::DimensionValue> dim_val;
+  if (tensor_shape_dim.has_dim_param()) {
+    dim_val = fbs::CreateDimensionValueDirect(builder, fbs::DimensionValueType_PARAM, 0, tensor_shape_dim.dim_param().c_str());
+  } else if (tensor_shape_dim.has_dim_value()) {
+    dim_val = fbs::CreateDimensionValueDirect(builder, fbs::DimensionValueType_VALUE, tensor_shape_dim.dim_value());
+  } else {
+    dim_val = fbs::CreateDimensionValueDirect(builder);
+  }
+
+  return fbs::CreateDimension(builder, dim_val, denotation);
+}
+
+static Status GetTensorShapeOrtFormat(flatbuffers::FlatBufferBuilder& builder,
+                                      const TensorShapeProto& tensor_shape_proto,
+                                      flatbuffers::Offset<fbs::Shape>& fbs_shape) {
+  std::vector<flatbuffers::Offset<fbs::Dimension>> dim;
+  dim.reserve(tensor_shape_proto.dim_size());
+  for (const auto& d : tensor_shape_proto.dim()) {
+    auto fbs_d = GetTensorDimensionOrtFormat(builder, d);
+    dim.push_back(fbs_d);
+  }
+  fbs_shape = fbs::CreateShapeDirect(builder, &dim);
+  return Status::OK();
+}
+
+static Status GetTensorTypeAndShapeOrtFormat(flatbuffers::FlatBufferBuilder& builder,
+                                             const TypeProto_Tensor& tensor_type_proto,
+                                             flatbuffers::Offset<fbs::TensorTypeAndShape>& fbs_tensor_type) {
+  flatbuffers::Offset<fbs::Shape> shape = 0;
+  if (tensor_type_proto.has_shape()) {
```

Comment:
Comment saying '0' is used for a null or missing shape would make the code easier to understand for someone who isn't familiar with the flatbuffers internals.